src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -150,6 +150,19 @@ internal bool TryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSeriali
                         ref reader);
                 }
 
+                if (CanBePolymorphic && options.ReferenceHandler != null)
```

Comment:
Could another approach be to intercept the call to the Object/JsonElementConverter (when the ref handling feature is on) and just read the $ref metadata directly?

Reply:
Yes, but we would have to create a copy of the reader that we would need to throw away if the payload doesn't met the required conditions (being an object, contain only the $ref property). Since this is an edge case, I think this approach is cleaner.